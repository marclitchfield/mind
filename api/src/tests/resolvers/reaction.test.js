import { applySetup, verifyMutations } from '../test-setup';
import gql from 'graphql-tag';

describe('Reaction tests', () => {
  applySetup();

  test('create reaction', () => {
    return verifyMutations('Reaction tests', 'post reaction', {
      mutations: [
        ({space}) => gql`mutation { Idea { subject:post_in_space(input:{sourceId:"${space.id}", title: "subject idea"}) { id } } }`,
        ({space}) => gql`mutation { Idea { response:post_in_space(input:{sourceId:"${space.id}", title: "response idea"}) { id } } }`,
        ({subject, response}) => gql`mutation { Reaction { reaction:post(input:{subjectIdeaId:"${subject.id}", responseIdeaId:"${response.id}", type:"answer"}) { id } } }`,
      ],
      query: ({reaction}) => gql`query {
        Reaction(id:"${reaction.id}") {
          type
          subject {
            title
            reactionsTo { type subject { title } response { title } }
            reactionsOf { type subject { title } response { title } }
          }
          response {
            title
            reactionsTo { type subject { title } response { title } }
            reactionsOf { type subject { title } response { title } }
          }
        }
      }`
    });
  });

  test('create reaction by person', () => {
    return verifyMutations('Reaction tests', 'create reaction by person', {
      mutations: [
        ({space}) => gql`mutation { Idea { subject:post_in_space(input:{sourceId:"${space.id}", title: "subject idea"}) { id } } }`,
        ({space}) => gql`mutation { Idea { response:post_in_space(input:{sourceId:"${space.id}", title: "response idea"}) { id } } }`,
        ({space}) => gql`mutation { Person { person:post_in_space(input:{sourceId:"${space.id}", title: "test person"}) { id } } }`,
        ({subject, response, person}) => gql`mutation { Reaction { 
          reaction:post(input:{sourcePersonId: "${person.id}", subjectIdeaId:"${subject.id}", responseIdeaId:"${response.id}" type:"answer"}) { id } } 
        }`,
      ],
      query: ({reaction}) => gql`query {
        Reaction(id:"${reaction.id}") {
          type
          subject { title }
          response { title }
          sourcePerson { title }
        }
      }`
    });
  });

  test('create reaction for event', () => {
    return verifyMutations('Reaction tests', 'create reaction for event', {
      mutations: [
        ({space}) => gql`mutation { Idea { subject:post_in_space(input:{sourceId:"${space.id}", title: "subject idea"}) { id } } }`,
        ({space}) => gql`mutation { Idea { response:post_in_space(input:{sourceId:"${space.id}", title: "response idea"}) { id } } }`,
        ({space}) => gql`mutation { Event { event:post_in_space(input:{sourceId:"${space.id}", title: "test event", datetime: "2011-12-12"}) { id } } }`,
        ({subject, response, event}) => gql`mutation { Reaction { 
          reaction:post(input:{eventId: "${event.id}", subjectIdeaId:"${subject.id}", responseIdeaId:"${response.id}" type:"answer"}) { id } } 
        }`,
      ],
      query: ({reaction}) => gql`query {
        Reaction(id:"${reaction.id}") {
          type
          subject { title }
          response { title }
          event { title }
        }
      }`
    });
  });

  test('remove reaction', () => {
    return verifyMutations('Reaction tests', 'remove reaction', {
      mutations: [
        ({space}) => gql`mutation { Idea { subject:post_in_space(input:{sourceId:"${space.id}", title: "subject idea"}) { id } } }`,
        ({space}) => gql`mutation { Idea { response:post_in_space(input:{sourceId:"${space.id}", title: "response idea"}) { id } } }`,
        ({space}) => gql`mutation { Event { event:post_in_space(input:{sourceId:"${space.id}", title: "test event", datetime: "2011-12-12"}) { id } } }`,
        ({space}) => gql`mutation { Person { person:post_in_space(input:{sourceId:"${space.id}", title: "test person"}) { id } } }`,
        ({subject, response, person, event}) => gql`mutation { Reaction { reaction:post(input:{
          sourcePersonId: "${person.id}", eventId: "${event.id}", subjectIdeaId:"${subject.id}", responseIdeaId:"${response.id}" type:"answer"}) { id } } 
        }`,
        ({reaction, subject, response, person, event}) => gql`mutation { Reaction { post(input:{
          id: "${reaction.id}", remove: true, sourcePersonId: "${person.id}", eventId: "${event.id}", subjectIdeaId:"${subject.id}", responseIdeaId:"${response.id}"}) { id } } 
        }`,
      ],
      query: ({subject, response}) => gql`query {
        i1:Idea(id:"${subject.id}") {
          reactionsOf { type }
          reactionsTo { type }
        }
        i2:Idea(id:"${response.id}") {
          reactionsOf { type }
          reactionsTo { type }
        }
      }`
    });
  });
});