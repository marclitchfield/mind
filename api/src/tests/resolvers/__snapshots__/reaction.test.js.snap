// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reaction tests create reaction 1`] = `
Object {
  "Reaction": Object {
    "__typename": "Reaction",
    "response": Object {
      "__typename": "Idea",
      "reactionsOf": Array [],
      "reactionsTo": Array [
        Object {
          "__typename": "Reaction",
          "response": Object {
            "__typename": "Idea",
            "title": "response idea",
          },
          "subject": Object {
            "__typename": "Idea",
            "title": "subject idea",
          },
          "type": "answer",
        },
      ],
      "title": "response idea",
    },
    "subject": Object {
      "__typename": "Idea",
      "reactionsOf": Array [
        Object {
          "__typename": "Reaction",
          "response": Object {
            "__typename": "Idea",
            "title": "response idea",
          },
          "subject": Object {
            "__typename": "Idea",
            "title": "subject idea",
          },
          "type": "answer",
        },
      ],
      "reactionsTo": Array [],
      "title": "subject idea",
    },
    "type": "answer",
  },
}
`;

exports[`Reaction tests create reaction by person 1`] = `
Object {
  "Reaction": Object {
    "__typename": "Reaction",
    "response": Object {
      "__typename": "Idea",
      "title": "response idea",
    },
    "sourcePerson": Object {
      "__typename": "Person",
      "title": "test person",
    },
    "subject": Object {
      "__typename": "Idea",
      "title": "subject idea",
    },
    "type": "answer",
  },
}
`;

exports[`Reaction tests create reaction for event 1`] = `
Object {
  "Reaction": Object {
    "__typename": "Reaction",
    "event": Object {
      "__typename": "Event",
      "title": "test event",
    },
    "response": Object {
      "__typename": "Idea",
      "title": "response idea",
    },
    "subject": Object {
      "__typename": "Idea",
      "title": "subject idea",
    },
    "type": "answer",
  },
}
`;

exports[`Reaction tests remove reaction 1`] = `
Object {
  "i1": Object {
    "__typename": "Idea",
    "reactionsOf": Array [],
    "reactionsTo": Array [],
  },
  "i2": Object {
    "__typename": "Idea",
    "reactionsOf": Array [],
    "reactionsTo": Array [],
  },
}
`;
